drivers/sensor/sbs_gauge/sbs_gauge.c

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (c) 2021 Leica Geosystems AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sbs_gauge
+
+#include <drivers/i2c.h>
+#include <init.h>
+#include <drivers/sensor.h>
+#include <sys/__assert.h>
+#include <string.h>
+#include <sys/byteorder.h>
+
+#include "sbs_gauge.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(sbs_gauge, CONFIG_SENSOR_LOG_LEVEL);
+
+static int sbs_cmd_reg_read(const struct device *dev, uint8_t reg_addr,
+				    int16_t *val)
+{
+	const struct sbs_gauge_config *const cfg = (struct sbs_gauge_config *const)dev->config;
+	uint8_t i2c_data[2];
+	int status;
+
+	status = i2c_burst_read(cfg->i2c_dev, cfg->i2c_addr, reg_addr,
+				i2c_data, 2);
+	if (status < 0) {
+		LOG_ERR("Unable to read register");
+		return -EIO;
```

Comment:
Masking the error messages like this might make it harder to debug problems; why not just let the i2c_burst_read() error code propagate to the caller (and be returned by the API function to the user of the driver)?

Reply:
agree, the read function is now returning the status of the i2c_burst_read()